plugins/block-dynamic-connection/src/dynamic_if.ts

```diff
@@ -11,9 +11,15 @@
 
 import * as Blockly from 'blockly/core';
 
-type DynamicIfBlock = Blockly.Block | (typeof Blockly.Blocks)['dynamic_if'];
-
-Blockly.Blocks['dynamic_if'] = {
+/** Type of a block that has the dynamicIfMixin. */
+type DynamicIfBlock = Blockly.Block & DynamicIfMixin;
+/* eslint-disable @typescript-eslint/no-empty-interface */
+/** This interface avoids a "circular reference" compile error. */
+interface DynamicIfMixin extends DynamicIfMixinType {}
+/* eslint-enable @typescript-eslint/no-empty-interface */
+type DynamicIfMixinType = typeof dynamicIfMixin;
+
+const dynamicIfMixin = {
```

Comment:
In core I've used `CONSTANT_CASE` for these mixins [because they're not supposed to be mutated](go/tsjs-style#identifiers-constants).  I don't know if there are other named block mixins in samples; if there are then it probably is best to be consistent with them, but if not then I recommend using the same convention as core.

Reply:
Done.